### PR TITLE
Fix form submit value patch generation

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/1166

#### Changes

- Remove `ids.dev_addr` from allowed field masks fort `set` requests
- Refactor `diff` to do deep object comparison 
- Add check for `boolean` values when merging devices from different sources
- Fix `mac_settings` in the device validation schema

#### Notes for Reviewers

The main issue in the device form was sending `ids.dev_addr` when updating device since the NS does not allow it and expects `dev_addr` to be in the `session` entry.

@rvolosatovs please check NS related changes

#### Release Notes

- Removed `ids.dev_addr` from allowed field masks for `/ttn.lorawan.v3.NsEndDeviceRegistry/Set`

### CODE REVIEW

File: pkg/ttnpb/field_mask_validation.go [hunk 1]
```diff
@@ -497,7 +497,6 @@ var AllowedFieldMaskPathsForRPC = map[string][]string{
 		"ids",
 		"ids.application_ids",
 		"ids.application_ids.application_id",
-		"ids.dev_addr",
```
Comment (rvolosatovs) [inline pkg/ttnpb/field_mask_validation.go]: @htdvisser  can you confirm that this is not passed by CLI anymore? If so, I'm good with it.
FTR, AS does not allow it to be set either.

Comment (htdvisser) [reply pkg/ttnpb/field_mask_validation.go]: The CLI uses `AllowedFieldMaskPathsForRPC` to determine what fields to set/get, so that should be fine.

Comment (johanstokking) [reply pkg/ttnpb/field_mask_validation.go]: Isn't this for setting ABP device addresses?

Comment (bafonins) [reply pkg/ttnpb/field_mask_validation.go]: > Isn't this for setting ABP device addresses?

The console sets it in `session.dev_addr`

Comment (johanstokking) [reply pkg/ttnpb/field_mask_validation.go]: Right

File: pkg/webui/lib/diff.js [hunk 2]
```diff
@@ -12,17 +12,26 @@
...
-    if (!ignorePaths.includes(field) && oldValue !== newValue) {
-      patch[field] = newValue
+/**
+ * Computes the structural differences (deep) between `original` and `updated` objects
+ * by applying any differences (add/remove/update).
+ * @param {Object} original - The original object.
+ * @param {Object} updated - The updated version of the `original` object.
+ * @param {Array} exclude - A list of field names that should not be included in the
+ * final diff.
+ * @returns {Object} - A new object represing the structural differences between `original` and `updated`.
```
Comment (kschiffer) [inline pkg/webui/lib/diff.js]:
```suggestion
 * @returns {Object} - A new object representing the structural differences between `original` and `updated`.
```

**Pull request merged**